.github/actions/use-ci-nexus-cache/action.yml

```diff
@@ -0,0 +1,51 @@
+---
+name: Use CI Nexus cache
+
+description: Sets up use of co-located pull-through cache for Maven artifacts via ~/.m2/settings.xml.
+
+inputs:
+  vault-url:
+    description: 'Vault URL to use.'
+    default: 'https://stage.vault.int.camunda.com'
+
+runs:
+  using: composite
+  steps:
+  - name: Create ~/.m2/settings.xml with cache configuration
+    shell: bash
+    run: |
+      cat <<EOF > ~/.m2/settings.xml
+      <?xml version="1.0" encoding="UTF-8"?>
+      <settings xmlns="http://maven.apache.org/SETTINGS/1.0.0"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://maven.apache.org/SETTINGS/1.0.0 http://maven.apache.org/xsd/settings-1.0.0.xsd">
+        <interactiveMode>false</interactiveMode>
+        <servers>
+          <server>
+            <id>camunda-nexus</id>
+            <username>\${env.NEXUS_USERNAME}</username>
+            <password>\${env.NEXUS_PASSWORD}</password>
+          </server>
+        </servers>
+        <mirrors>
+          <mirror>
+            <id>camunda-nexus</id>
+            <mirrorOf>*</mirrorOf>
+            <name>Camunda Nexus</name>
+            <url>http://repository-ci-camunda-cloud.nexus:8081/content/groups/internal/</url>
+          </mirror>
+        </mirrors>
+      </settings>
+      EOF
+
```

Comment:
maybe add a comment to mention that the secrets are pulled via a SA of the kubernetes pod limited to the exact secrets.